Add Download Metadata Action

mzagaja opened the pull request:
This commit adds an action for downloading metadata called downloadMetadata and making it available in the template.

Resolves MAPC/datacommon#63

mzagaja committed:
Add Download Metadata Action

This commit adds an action for downloading metadata called downloadMetadata and making it available in the template.

Resolves MAPC/datacommon#63

ericyoungberg commented on app/controllers/datasets.js:
```diff
@@ -148,6 +147,37 @@ export default class extends Controller {
+
+    const link = document.createElement('a');
+    link.setAttribute('href', encoded);
+    link.setAttribute('download', `metadata.csv`);
```
LGTM besides we need the table name, e.g. `cbecs_table_thing-metadata.csv`

mzagaja committed:
Add Data Title to File Name

This adds the title of the data chart to the file name in order to make it easier for users to recall which data set the metadata is for.
